src/main/java/core/basesyntax/Rectangle.java

```diff
@@ -27,4 +27,30 @@ public Integer getLength() {
     public String getColor() {
         return color;
     }
+
+    @Override
+    public boolean equals(Object object) {
+        if (this == object) {
+            return true;
+        }
+        if (!(object instanceof Rectangle) || object == null) {
```

Comment:
what the difference between instanceof and getClass()?

Reply:
I see my mistake, if i will create subclass from Rectangle, object of this subclass will not equals to Rectangle object and Rectangle will be equal to child-class object. For strict type matching I need to use getClass()